sql/core/src/main/scala/org/apache/spark/sql/execution/dynamicpruning/PartitionPruning.scala

```diff
@@ -65,7 +65,7 @@ object PartitionPruning extends Rule[LogicalPlan] with PredicateHelper {
           case fs: HadoopFsRelation =>
             val partitionColumns = AttributeSet(
               l.resolve(fs.partitionSchema, fs.sparkSession.sessionState.analyzer.resolver))
-            if (resExp.references.subsetOf(partitionColumns)) {
+            if (!resExp.references.isEmpty && resExp.references.subsetOf(partitionColumns)) {
```

Comment:
`resExp.references.nonEmpty`?

Reply:
Does the `DataSourceV2ScanRelation` also need this check?

Reply:
HadoopFsRelation, HiveTableRelation, DataSourceV2ScanRelation all need this check, and now I filter the literal at the beginning.

Is such code reasonable? Or should I handle it in findExpressionAndTrackLineageDown? Or should it be handled in three case statements?

Thank you for your review

Reply:
Could we fix `PredicateHelper.findExpressionAndTrackLineageDown`?

Reply:
Is this ok now?

Reply:
LGTM.